server.js

```diff
@@ -77,9 +77,11 @@ app.post('/api/v1/waves/', (request, response) => {
 app.delete('/api/v1/sources/:id', (request, response) => {
   const { id } = request.params;
 
-  database('sources').where({SOURCE_ID: id}).del()
+  database('waves').where({SOURCE_ID: id}).del().then(() => {
```

Comment:
Here is where I called delete on all waves whose foreign Ids point to the source we are deleting.